### DISCUSSION TITLE
_app.tx in NextJS14

### DISCUSSION THREAD

**Discussion by damianlluch** [vercel/next.js#62392] [Help] [answered]
### Summary

Do you have any idea how to create an _app.tsx file in NextJS14 ? I haven't found a document and I need to define a provider for web3-react.js ?

Thanks

**Comment by makibat** [comment]
Hey!

You could create a `providers.tsx` Client Component that wraps the third-party provider at the root of your app directory. The reason why it needs to be a Client Component is that React Server Components don't support creating or consuming context directly, therefore, it will result in an error.

``` js
'use client';

import { Provider } from ...

export function Providers({ children }) {
  return (
    <Provider>
      {children}
    </Provider>
  );
}
```

And then just import `<Providers />` in your root layout

``` js
import { Providers } from './providers';

export default function RootLayout({ children }) {
  return (
    <html>
      <body>
        <Providers>{children}</Providers>
      </body>
    </html>
  );
}
```

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Do you mean an _app file within App Router? The App Router stack doesn't use _app anymore. Rather, you have `app/layout.tsx` referrer to as RootLayout, where you can inject various things into the React tree, such as Providers, which themselves can be client components, even though the Root layout stays a server component.

**Reply by damianlluch** [reply]
Hello, thank you, yes I have done it this way, solved. Thanks
